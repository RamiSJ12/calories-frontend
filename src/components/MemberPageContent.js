import {
  Card,
  CardActionArea,
  CardActions,
  CardContent,
  CardMedia,
  makeStyles,
  TextField,
  Typography,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import commonStyles from '../utils/commonStyles';

const useStyles = makeStyles({
  root: {
    width: '90%',
    maxWidth: 500,
    boxShadow: 'none',
    margin: '30px auto 0',
  },
  media: {
    height: 140,
  },
});

const MemberPageContent = ({
  name,
  image,
  info,
  changeHandler,
}) => {
  const classes = useStyles();
  const commonClasses = commonStyles();
  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia
          className={classes.media}
          image={image}
        />
        <CardContent>
          <Typography gutterBottom color="textSecondary" variant="h5" component="h2">
            {name}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {info}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <TextField
          onChange={e => changeHandler(name, e.target.value)}
          className={commonClasses.darkText}
          fullWidth
          value=""
          name={name}
          label={name}
          required
          variant="outlined"
          helperText="Should be in Gram"
          margin="dense"
          type="number"
        />
      </CardActions>
    </Card>
  );
};

MemberPageContent.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  info: PropTypes.string.isRequired,
  changeHandler: PropTypes.func.isRequired,
};

export default MemberPageContent;
