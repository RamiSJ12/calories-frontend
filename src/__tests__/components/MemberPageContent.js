import renderer from 'react-test-renderer';
import MemberPageContent from '../../components/MemberPageContent';

describe('MemberPageContent component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<MemberPageContent
      name="Carbohydrates"
      image="fajdjf"
      info="Some text"
      value="25"
      changeHandler={() => ({})}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
