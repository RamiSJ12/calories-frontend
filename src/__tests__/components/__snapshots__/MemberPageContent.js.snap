// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberPageContent component should match snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root makeStyles-media-3"
      style={
        Object {
          "backgroundImage": "url(\\"fajdjf\\")",
        }
      }
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        Carbohydrates
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Some text
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root makeStyles-cardActionRoot-2 MuiCardActions-spacing"
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
      >
        Carbohydrates
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input makeStyles-darkText-6 MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          min={0}
          name="Carbohydrates"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="number"
          value="25"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-27 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-29 PrivateNotchedOutline-legendNotched-30"
          >
            <span>
              Carbohydrates
               *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required MuiFormHelperText-marginDense"
      >
        Should be in Gram
      </p>
    </div>
  </div>
</div>
`;
